### PR TITLE
[incubator/buzzfeed-sso] add suport for v2.0.0

### PR DESCRIPTION
**Pull request by cameronattard**
#### What this PR does / why we need it:
Adds support for `buzzfeed/sso` `v2.0.0`.

#### Special notes for your reviewer:
This is not backwards compatible with `v1.x.x`. Changes are documented [here](https://github.com/buzzfeed/sso/releases/tag/v2.0.0).

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

@darioblanco @komljen

### CODE REVIEW

File: incubator/buzzfeed-sso/values.yaml [hunk 1]
```diff
@@ -36,7 +36,11 @@ auth:
 
 proxy:
   annotations: {}
-  extraEnv: []
+  extraEnv:
+    - name: STATSD_HOST
```
Comment (komljen) [inline incubator/buzzfeed-sso/values.yaml:40]: STATSD_HOST -> METRICS_STATSD_HOST

File: incubator/buzzfeed-sso/values.yaml [hunk 2]
```diff
@@ -36,7 +36,11 @@ auth:
...
+      value: localhost
+    - name: STATSD_PORT
```
Comment (komljen) [inline incubator/buzzfeed-sso/values.yaml:42]: STATSD_PORT -> METRICS_STATSD_PORT

**Review by komljen — APPROVED** [review]
Just two more env variable updates for metrics according to https://github.com/buzzfeed/sso/releases/tag/v2.0.0 and it is ok I think.

**Comment by cameronattard** [comment]
@komljen it seems the environment variables were only updated for the `auth` component - the `proxy` ones remain the same: https://github.com/buzzfeed/sso/blob/master/internal/proxy/options.go#L95

**Comment by komljen** [comment]
> @komljen it seems the environment variables were only updated for the `auth` component - the `proxy` ones remain the same: https://github.com/buzzfeed/sso/blob/master/internal/proxy/options.go#L95

All good then.

**Pull request merged**